File: simple-statemachine-with-persistence/uising-statemachine-data-jpa/simple-statemachine-java/src/main/java/com/example/statemachine/simple/doument/config/JpaPersisterConfig.java
package com.example.statemachine.simple.doument.config;

import com.example.statemachine.simple.doument.DocumentEvent;
import com.example.statemachine.simple.doument.DocumentState;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.statemachine.data.jpa.JpaPersistingStateMachineInterceptor;
import org.springframework.statemachine.data.jpa.JpaStateMachineRepository;
import org.springframework.statemachine.persist.StateMachineRuntimePersister;

@Configuration
public class JpaPersisterConfig {

  @Bean
  public StateMachineRuntimePersister<DocumentState, DocumentEvent, String> stateMachineRuntimePersister(
    JpaStateMachineRepository jpaStateMachineRepository) {
    return new JpaPersistingStateMachineInterceptor<>(jpaStateMachineRepository);
  }

}

File: simple-statemachine-with-persistence/uising-statemachine-data-jpa/simple-statemachine-kotlin/src/main/kotlin/com/example/statemachine/simple/document/config/JpaPersisterConfig.kt
package com.example.statemachine.simple.document.config

import com.example.statemachine.simple.document.DocumentEvent
import com.example.statemachine.simple.document.DocumentState
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.statemachine.data.jpa.JpaPersistingStateMachineInterceptor
import org.springframework.statemachine.data.jpa.JpaStateMachineRepository
import org.springframework.statemachine.persist.StateMachineRuntimePersister

@Configuration
class JpaPersisterConfig {
  @Bean
  fun stateMachineRuntimePersister(
    jpaStateMachineRepository: JpaStateMachineRepository?
  ): StateMachineRuntimePersister<DocumentState, DocumentEvent, String> {
    return JpaPersistingStateMachineInterceptor(
      jpaStateMachineRepository
    )
  }
}
